src/models/request-status.ts

```diff
@@ -4,3 +4,9 @@ export enum RequestStatus {
   COMPLETED = 2,
   FAILED = 3,
 }
+
+export class InvalidRequestStatusError extends Error {
```

Comment:
We have plans to introduce a proper error system based on https://github.com/ceramicnetwork/js-errors-utils.  Until we do that, I'd rather keep our errors simple and hierarchy free, so they're easy to update to the new system later.  Can you just use raw `Error` for now?

Reply:
While I agree on unification, I would like to leave it here as it is. Thanks to `never` argument it exhaustively checks if all the enum variants are handled.

Attached for reference: https://github.com/ceramicnetwork/js-errors-utils/issues/2

Reply:
wow, fascinating.  What if you removed the `default` case from the switch statement, and put the throw after the switch statement - wouldn't that have the same effect?

Reply:
Not at all. That's the point. With `throw` in `default` case, the compiler does not complain when you add a new variant to enum and forget to handle it.

Reply:
If you have no default case, the compiler won't warn you if you don't handle an enum variant case?  That's pretty disappointing